types/json.go

```diff
@@ -0,0 +1,66 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+)
+
+// CheckDuplicateKey checks duplicate keys in JSON
+func CheckDuplicateKey(d *json.Decoder, path []string) error {
```

Comment:
This was moved outside `ecocredit/client` so that it could be reused in `ecocredits/client/marketplace` and `ecocredit/client/basket` without import issues. Also might be helpful in other parts of the code in the future.